### PR TITLE
Week 2 - Initial Functionality

### PR DESCRIPTION
**Pull request by SiftingSands**
Demonstrate read and write capability with simple song cross fading. Four example tracks are included with linear and sigmoid transitions.

### CODE REVIEW

File: src/analysis/explore.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+print('File: {}'.format(filename))
+
+x, sr = librosa.load(AUDIO_DIR + filename, sr=None, mono=True)
+print('Duration: {:.2f}s, {} samples'.format(x.shape[-1] / sr, x.size))
+time = np.arange(0,len(x),1)/sr
+
+snippet_time = 5 # seconds
+snippet_length = snippet_time * sr # number of samples
+
+# Calculate Mel Spectrogram
```
Comment (mrunfeldt) [inline src/analysis/explore.py:28]: MEL is optimized for human language. My concern here is that you are going to lose valuable spectral information from the music. I recommend just using a regular FFT to create your spectrograms and not do any initial filtering (like MEL) before hand. You can look at the data across a few songs to define the spectral range (min and max) and resolution (bin size)

File: src/analysis/explore.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+nfft = 2048
+hop_length = 512
```
Comment (mrunfeldt) [inline src/analysis/explore.py:30]: It's good to get in the habit of defining all of your parameters at the top/beginning of your code. If you have time later, you can then pass them with a param file or as a command input.

File: src/model/crossfade_simple.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+time1 = np.arange(0,len(x1),1)/sr1
+
+x2, sr2 = librosa.load(AUDIO_DIR + selected_files[1], sr=None, mono=True)
+print('Song 2 Duration: {:.2f}s, {} samples'.format(x2.shape[-1] / sr2, 
+      x2.size))
+time2 = np.arange(0,len(x2),1)/sr2
+
+# Trim the songs to the beginning and end
+if sr1 != sr2:
+    raise ValueError('Sampling rates between songs must be identical')
```
Comment (mrunfeldt) [inline src/model/crossfade_simple.py]: Error catching, nice!

Comment (mrunfeldt) [reply src/model/crossfade_simple.py]: If you have time, you might want to make a function to check sampling rate before you even load the songs. Not a big priority right now, though

**Review by mrunfeldt — CHANGES_REQUESTED** [review]
You probably don't want to use to use MEL spectrogram. Other than that, LGTM (looks good to me)

**Review by mrunfeldt — APPROVED** [review]

File: src/notebooks/nsynth_interpolation.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+###############################################################################
+
+def LinearFade(length):
+    fadein = np.linspace(0, 1, length).reshape(1, -1, 1)
+    return fadein
+
+###############################################################################
+    
+def HannFade(length):
+    fadein = (0.5 * (1.0 - np.cos(3.1415 * np.arange(length) /
```
Comment (mrunfeldt) [inline src/notebooks/nsynth_interpolation.py:22]: math.pi

Comment (SiftingSands) [reply src/notebooks/nsynth_interpolation.py:22]: Got it!

**Pull request merged**